File: YoutubeClone/DTO/SubscriptionSummary.cs
namespace YoutubeClone.DTO
{
    public class SubscriptionSummary
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string UserProfilePictureUrl { get; set; } = string.Empty;
        public Guid ChannelId { get; set; }
        public string ChannelName { get; set; } = string.Empty;
        public string ChannelThumbnailUrl { get; set; } = string.Empty;
    }
}

File: YoutubeClone/ClientApp/src/api/models/subscription-summary.ts
export interface SubscriptionSummary {
    id: string;
    userId: string;
    userName: string;
    userProfilePictureUrl: string;
    channelId: string;
    channelName: string;
    channelThumbnailUrl: string;
}

export function mapJsonToSubscriptionSummary(json: any): SubscriptionSummary {
    return {
        id: json.id,
        userId: json.userId,
        userName: json.userName,
        userProfilePictureUrl: json.userProfilePictureUrl,
        channelId: json.channelId,
        channelName: json.channelName,
        channelThumbnailUrl: json.channelThumbnailUrl
    }
}